packages/discord.js/src/managers/AutoModerationRuleManager.js

```diff
@@ -0,0 +1,292 @@
+                mention_total_limit: triggerMetadata.mentionTotalLimit,
+              },
+        actions: actions.map(action => ({
+          type: action.type,
+          metadata: {
+            duration_seconds: action.metadata?.durationSeconds,
+            channel_id:
+              typeof action.metadata?.channel === 'undefined'
+                ? undefined
+                : this.guild.channels.resolveId(action.metadata.channel),
+          },
+        })),
+        enabled,
+        exempt_roles: exemptRoles?.map(exemptRole => this.guild.roles.resolveId(exemptRole)),
+        exempt_channels: exemptChannels?.map(exemptChannel => this.guild.channels.resolveId(exemptChannel)),
+      },
+      reason,
+    });
+
+    return this._add(data);
+  }
+
+  // TODO: discord-api-types enum
+  /**
+   * Options used to edit an auto moderation rule.
+   * @typedef {Object} AutoModerationRuleEditOptions
+   * @property {string} [name] The name of the auto moderation rule
+   * @property {number} [eventType] The event type of the auto moderation rule
+   * @property {AutoModerationTriggerMetadataOptions} [triggerMetadata] The trigger metadata of the auto moderation rule
+   * @property {AutoModerationActionOptions[]} [actions]
+   * The actions that will execute when the auto moderation rule is triggered
+   * @property {boolean} [enabled] Whether the auto moderation rule should be enabled
+   * @property {Collection<Snowflake, Role>|RoleResolvable[]} [exemptRoles]
+   * The roles that should not be affected by the auto moderation rule
+   * @property {Collection<Snowflake, GuildChannel|ThreadChannel>|GuildChannelResolvable[]} [exemptChannels]
+   * The channels that should not be affected by the auto moderation rule
+   * @property {string} [reason] The reason for creating the auto moderation rule
+   */
+
+  /**
+   * Edits an auto moderation rule.
+   * @param {AutoModerationRuleResolvable} autoModerationRule The auto moderation rule to edit
+   * @param {AutoModerationRuleEditOptions} options Options for editing the auto moderation rule
+   * @returns {Promise<AutoModerationRule>}
+   */
+  async edit(
+    autoModerationRule,
+    { name, eventType, triggerMetadata, actions, enabled, exemptRoles, exemptChannels, reason },
+  ) {
+    const autoModerationRuleId = this.resolveId(autoModerationRule);
+
+    const data = await this.client.rest.patch(
+      // TODO: discord-api-types route
+      `/guilds/${this.guild.id}/auto-moderation/rules/${autoModerationRuleId}`,
+      {
+        body: {
+          name,
+          event_type: eventType,
+          trigger_metadata:
+            typeof triggerMetadata === 'undefined'
+              ? undefined
+              : {
+                  keyword_filter: triggerMetadata.keywordFilter,
+                  presets: triggerMetadata.presets,
+                  allow_list: triggerMetadata.allowList,
+                  mention_total_limit: triggerMetadata.mentionTotalLimit,
+                },
```

Comment:
```suggestion
            triggerMetadata && {
                  keyword_filter: triggerMetadata.keywordFilter,
                  presets: triggerMetadata.presets,
                  allow_list: triggerMetadata.allowList,
                  mention_total_limit: triggerMetadata.mentionTotalLimit,
                },
```
this would have the same affect and is slightly neater IMO